examples/llm/inference/README.md

```diff
@@ -0,0 +1,136 @@
+```
+my_hf_model/
+  config.json
+  merges.txt
+  pytorch_model.bin
+  special_tokens_map.json
+  tokenizer.json
+  tokenizer_config.json
+  vocab.json
+```
+which can be loaded with standard HF utilities like `AutoModelForCausalLM.from_pretrained('my_hf_model')`.
+
+You can also pass object store URIs for both `--composer_path` and `--hf_output_path` to easily convert checkpoints stored in S3, OCI, etc.
+
+### IMPORTANT NOTE:
+If you trained and saved a custom HF model such as `MosaicGPT`, then in any external inference codebase, you need to import and register the new model class and config before auto classes like `AutoModel` will work. For example:
+
+```python
+# MosaicGPT, MosaicGPTConfig source code live in this repo
+# pip install mosaicml-examples
+
+from examples.llm import MosaicGPT, MosaicGPTConfig
+
+AutoConfig.register('mosaic_gpt', MosaicGPTConfig)
+AutoModelForCausalLM.register(MosaicGPTConfig, MosaicGPT)
+
+model = AutoModelForCausalLM.from_pretrained('my_hf_model')
+```
+
+(Coming soon) we will add the ability to save custom model source code within the HF folder, which will remove the need for this step!
+
+
+## Interactive generation with `model.generate(...)`
+
+To make it easy to inspect the generations produced by your HF model, we include a script `hf_generate.py` that allows you to run custom prompts through your HF model, like so:
+
+```bash
+python hf_generate.py \
+    --name_or_path gpt2 \
+    --temperature 1.0 \
+    --top_p 0.95 \
+    --top_k 50 \
+    --seed 1 \
+    --max_new_tokens 256 \
+    --prompts \
+      "The answer to life, the universe, and happiness is" \
+      "MosaicML is an ML training efficiency starup that is known for" \
+      "My name is" \
+      "The best 5 cities to visit in Europe are"
+```
+
+which will produce output:
+
+```bash
+Loading HF model...
+n_params=124439808
+
+Loading HF tokenizer...
+/mnt/workdisk/abhi/examples/examples/llm/inference/hf_generate.py:89: UserWarning: pad_token_id is not set for the tokenizer. Using eos_token_id as pad_token_id.
+  warnings.warn(
+
+Generate kwargs:
+{'max_new_tokens': 256, 'temperature': 1.0, 'top_p': 0.95, 'top_k': 50, 'use_cache': True, 'do_sample': True, 'eos_token_id': 50256}
+
+Moving model and inputs to device=cuda and dtype=torch.bfloat16...
+
+Tokenizing prompts...
+NOT using autocast...
+Warming up...
+Setting `pad_token_id` to `eos_token_id`:50256 for open-end generation.
+Generating responses...
+Setting `pad_token_id` to `eos_token_id`:50256 for open-end generation.
+##################################################################
+The answer to life, the universe, and happiness is to love.
+
+...
+##################################################################
+MosaicML is an ML training efficiency starup that is known for its efficient and efficient strategies.
+
+...
+##################################################################
+My name is Eulogy Jackson, and I am the mother of nine children.
```

Comment:
Just in case this is a real person or something, can we change this example out?

Reply:
Will change the prompt.